moduleroot/appveyor.yml

```diff
@@ -0,0 +1,58 @@
+<%
+# Workaround to support boolean supports_windows and hash+array supported_oss
+# settings until one form is no longer used
+supports_windows = (
+  @configs['supports_windows'] == true \
+  || (
+    @configs['supported_oss'].is_a?(Hash) \
+    && @configs['supported_oss'][@configs[:puppet_module]].is_a?(Array) \
+    && @configs['supported_oss'][@configs[:puppet_module]].include?('windows')
+  )
+)
+
+if supports_windows -%>
+version: 1.1.x.{build}
+skip_commits:
+  message: /^\(?doc\)?.*/
+clone_depth: 10
+init:
+- SET
+- 'mkdir C:\ProgramData\PuppetLabs\code && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\facter && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\hiera && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\puppet\var && exit 0'
+environment:
+  matrix:
+<% @configs['matrix'].each do |matrix| -%>
+  - PUPPET_GEM_VERSION: <%= matrix['PUPPET_GEM_VERSION'] %>
+    RUBY_VER: <%= matrix['RUBY_VER'] %>
+<% end -%>
+<% if @configs['matrix_extras'] -%>
+<%   @configs['matrix_extras'].each do |extra| -%>
+  - PUPPET_GEM_VERSION: <%= extra['PUPPET_GEM_VERSION'] %>
+    RUBY_VER: <%= extra['RUBY_VER'] %>
+<%   end -%>
+<% end -%>
+matrix:
+  fast_finish: true
+install:
+- SET PATH=C:\Ruby%RUBY_VER%\bin;%PATH%
+- <%= @configs['appveyor_bundle_install'] %>
+- type Gemfile.lock
+build: off
+test_script:
+- bundle exec puppet -V
+- ruby -v
+<% @configs['test_script'].each do |script| -%>
+- <%= script %>
+<% end -%>
+notifications:
+- provider: Email
+  to:
+  - nobody@nowhere.com
+  on_build_success: false
+  on_build_failure: false
+  on_build_status_changed: false
+<% else -%>
+# Appveyor is only enabled on modules that support windows
```

Comment:
The condition of supported OSs (linux and/or windows) is not something built-in to modulesync so it needs ERB templating to present different yaml entries. config_defaults.yaml defines the files to be managed, one of which is appveyor.yml. It is not possible to conditionalize the management of appveyor.yml, only define the defaults. And the template itself cannot afaik say "actually I should not be created" only "I should be empty".
Options:
1. appveyor.yml exists in all repos and we manually enable/disable testing (one time cost per module, imho best option)
1. appveyor.yml exists in all repos and we test on all modules (what is testing apt in appveyor supposed to do? This doesn't make sense)
1. appveyor.yml key only exists in $module/.sync.yml and we have the defaults in :global in config_defaults.yml (this is an ugly hack)
1. modulesync is expanded to have templates be conditionally managed (complex, low return, and don't want to wait for this)